lib/wibox/container/margin.lua

```diff
@@ -95,20 +95,30 @@ end
 
 --- Set all the margins to val.
 -- @property margins
--- @tparam number val The margin value
+-- @tparam number|table val The margin value. It can be a number or a table with
+--  the *left*/*right*/*top*/*bottom* keys.
 
 function margin:set_margins(val)
-    if self._private.left   == val and
+    if type(val) == "number" and
+       self._private.left   == val and
        self._private.right  == val and
        self._private.top    == val and
        self._private.bottom == val then
         return
```

Comment:
Can you update this to catch "nothing changed" and not emit the signal in that case? For example:
```lua
function margin:set_margins(val)
    if type(val) == "table" then
        self:set_left(assert(val.left))
        etc...
    else
       existing code here
    end
end
```
(The `assert` would catch tables without the necessary entries; that way one gets an error with a better traceback than "something went wrong while drawing").

Reply:
I made the table more permissive instead (like for the `c:geometry()` and other similar APIs.